rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/widget/OneScanCovidRDTBarcodeFactoryTest.java

```diff
@@ -110,6 +117,32 @@ public void testOnActivityResultShouldPopulateCorrectData() throws Exception {
         Mockito.verify(formFragment).setMoveBackOneStep(true);
     }
 
+    @Test
+    public void testLaunchBarcodeScannerShouldVerifyActivityStartup() throws Exception {
+        MaterialEditText materialEditText = Mockito.mock(MaterialEditText.class);
+        InputMethodManager inputManager = Mockito.mock(InputMethodManager.class);
+        Activity activity = Mockito.mock(Activity.class);
+        Mockito.when(activity.getSystemService(ArgumentMatchers.anyString())).thenReturn(inputManager);
+        Whitebox.invokeMethod(oneScanCovidRDTBarcodeFactory, "launchBarcodeScanner", activity, materialEditText, "");
+        Mockito.verify(activity).startActivityForResult(ArgumentMatchers.any(Intent.class), ArgumentMatchers.anyInt());
+    }
+
+    @Test
+    public void testGetBarcodeValsAsCSVShouldReturnCorrectDisplayValue() throws Exception {
+        Barcode barcode = new Barcode();
+        barcode.displayValue = barcodeVals;
+        Intent intent = new Intent();
+        intent.putExtra(JsonFormConstants.BARCODE_CONSTANTS.BARCODE_KEY, barcode);
+        String result = Whitebox.invokeMethod(oneScanCovidRDTBarcodeFactory, "getBarcodeValsAsCSV", intent);
+        Assert.assertEquals(barcodeVals, result);
+    }
+
+    @Test
+    public void testSplitCSVShouldReturnCorrectArrays() throws Exception {
+        String[] result = Whitebox.invokeMethod(oneScanCovidRDTBarcodeFactory, "splitCSV", barcodeVals);
+        Assert.assertArrayEquals(new String[]{VAL_0, DATE, VAL_2, VAL_3, SENSOR_TRIGGERED}, result);
```

Comment:
Why not use `barcodeVals.split(",")` here?

Reply:
fixed.